master/pkg/model/agent.go

```diff
@@ -80,3 +81,32 @@ type AgentStats struct {
 	AgentID      string `db:"agent_id"`
 	Slots        int    `db:"slots"`
 }
+
+// ContainerSummary summarizes a container.
+type ContainerSummary struct {
+	ID           cproto.ID       `json:"id"`
+	State        cproto.State    `json:"state"`
+	Devices      []device.Device `json:"devices"`
+	AllocationID AllocationID    `json:"allocation_id"`
+	TaskID       TaskID          `json:"task_id"`
+	JobID        JobID           `json:"job_id"`
+}
+
+// ToProto converts a ContainerSummary to its protobuf representation.
+func (c *ContainerSummary) ToProto() *containerv1.Container {
+	if c == nil {
+		return nil
+	}
+	var devices []*devicev1.Device
+	for _, d := range c.Devices {
+		devices = append(devices, d.Proto())
+	}
```

Comment:
It's totally fine if you wanna leave this how it is, but I like to define a new type that's a slice and define a `Proto()` method on that slice type, [like with `Groups`](https://github.com/determined-ai/determined/blob/3511abf5cc2a/master/pkg/model/user_group.go#L27).

Reply:
Neat

Reply:
Is this for improving code quality, so all device slices has a Proto() instead of repeating the loops everywhere?

Reply:
Yeah, that exactly my reason.